src/AutoRest.CSharp/Common/Input/InputTypes/Serialization/TypeSpecInputDateTimeTypeConverter.cs

```diff
@@ -65,5 +66,30 @@ public static InputDateTimeType CreateDateTimeType(ref Utf8JsonReader reader, st
             }
             return dateTimeType;
         }
+
+        public override void Write(Utf8JsonWriter writer, InputDateTimeType value, JsonSerializerOptions options)
+            => writer.WriteObjectOrReference<InputDateTimeType>(value, options, _referenceHandler.CurrentResolver, WriteInputDateTimeType);
+
+        private static void WriteInputDateTimeType(Utf8JsonWriter writer, InputDateTimeType value, JsonSerializerOptions options)
+        {
+            writer.WriteStartObject();
+
+            // kind
+            writer.WriteString("kind", UtcDateTimeKind); // TODO -- currently the two different kinds have exactly the same properties, therefore here we no longer have the ability to distinguish
```

Comment:
what's the plan for this?

Reply:
it is possible that we make this class abstract, and make UtcDateTimeType and OffsetDateTimeType to derive from it, like what we have done for `InputType`.

Reply:
And I think we should also do the same in MTG.